packages/caravan-fees/src/types.ts

```diff
@@ -0,0 +1,444 @@
+ *
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki
+ */
+export interface UTXO {
+  /** The transaction ID of the UTXO in reversed hex format (big-endian). */
+  txid: string;
+
+  /** The output index of the UTXO in its parent transaction. */
+  vout: number;
+
+  /** The value of the UTXO in satoshis. */
+  value: Satoshis;
+
+  /**
+   * The full previous transaction in hexadecimal format.
+   * Required for non-segwit inputs in PSBTs.
+   */
+  prevTxHex?: string;
+
+  /**
+   * The witness UTXO information for segwit transactions.
+   * Required for segwit inputs in PSBTs.
+   */
+  witnessUtxo?: {
+    script: Buffer;
+    value: number;
+  };
+}
+
+/**
+ * Configuration options for the TransactionAnalyzer.
+ */
+export interface AnalyzerOptions {
+  /**
+   * The Bitcoin network to use (mainnet, testnet, or regtest).
+   */
+  network: Network;
+
+  /**
+   * The target fee rate in satoshis per vbyte that the user wants to achieve.
+   * This is used to determine if fee bumping is necessary and to calculate
+   * the new fee for RBF or CPFP.
+   */
+  targetFeeRate: number;
+
+  /**
+   * The absolute fee of the original transaction in satoshis.
+   * This is used as the basis for fee calculations and comparisons.
+   */
+  absoluteFee: string | number;
+
+  /**
+   * An array of Unspent Transaction Outputs (UTXOs) that are available
+   * for fee bumping. These are potential inputs that can be added to
+   * a replacement transaction in RBF, or used to create a child transaction
+   * in CPFP.
+   */
+  availableUtxos: UTXO[];
+
+  /**
+   * The index of the change output in the transaction, if known.
+   * This helps identify which output is the change and can be
+   * adjusted for fee bumping in RBF scenarios.
+   */
+  changeOutputIndex?: number;
+
+  /**
+   * The incremental relay fee in satoshis per vbyte. This is the minimum
+   * fee rate increase required for nodes to accept a replacement transaction.
+   * It's used in RBF calculations to ensure the new transaction meets
+   * network requirements.
+   * Default value in Bitcoin Core is 1 sat/vbyte.
+   * @see https://github.com/bitcoin/bitcoin/blob/master/src/policy/fees.h
+   */
+  incrementalRelayFee?: string | number;
+
+  /**
+   * The number of signatures required in a multisig setup.
+   * This is used to estimate transaction size more accurately
+   * for multisig transactions.
+   */
+  requiredSigners: number;
+
+  /**
+   * The total number of signers in a multisig setup.
+   * This is used along with requiredSigners to estimate
+   * transaction size more accurately for multisig transactions.
+   */
+  totalSigners: number;
+
+  /**
+   * The type of Bitcoin address used (e.g., P2PKH, P2SH, P2WPKH, P2WSH).
+   * This is used to determine the input size for different address types
+   * when estimating transaction size.
+   */
+  addressType: string;
```

Comment:
use types from @caravan/bitcoin for consistency. `MultisigAddressType`

Reply:
changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/8da49623b7f7ede0c72ea4fd31a4847d0811293e#diff-593f8d602f801c56261ed7a8dc941947102f6a7b26c483c5ee8cf9c293fffb33R444)